### ISSUE TITLE
A sort order was already defined.

### ISSUE THREAD

**Issue by preyesm** [realm/realm-java#6371]
Hi, I have a small doubt. My app crashed with the error as shown above. I have realm query of all matching rows and i need to show them under two different headers depending on the time,
```upcomingReminders = realmQuery.greaterThan("reminderTime", System.currentTimeMillis() / 1000).sort("reminderTime", DESCENDING).findAll();
        pastReminders = realmQuery.lessThan("reminderTime", System.currentTimeMillis() / 1000).sort("reminderTime", DESCENDING).findAll();```
so, my question is can we not use the same realm query and get two different result set.

**Comment by cmelchior** [comment]
The exception says that ever created `realmQuery` already called sort on it and calling sort again would just override the first one. So we throw an exception.

So either stop calling sort the first time or you can also create an intermediate query result this way:

```
pastReminders = realmQuery
  .lessThan("reminderTime", System.currentTimeMillis() / 1000)
  .findAll()
  .where()
  .sort("reminderTime", DESCENDING)
  .findAll();
```

**Comment by preyesm** [comment]
@cmelchior basically, i need two result sets from same realm query, instead of creating two query set, i thought i shall reuse the same query, I think I missed it earlier. please check the code below.
```
upcomingReminders = realmQuery.greaterThan("reminderTime", System.currentTimeMillis() / 1000).sort("reminderTime", DESCENDING).findAll();
        pastReminders = realmQuery.lessThan("reminderTime", System.currentTimeMillis() / 1000).sort("reminderTime", DESCENDING).findAll();
```

**Comment by cmelchior** [comment]
Yes, that doesn't work. Think of RealmQuery as a builder, so pastReminders would contain all the predicates you added in `upcommingReminders`. So even if sort didn't threw an exception, it would still be the wrong query.

**Comment by preyesm** [comment]
Ideally, maintaining a copy of realm query is the solution I believe. Thanks @cmelchior

**Comment by cmelchior** [comment]
Yes, right now you would need to manually create two queries, a better solution for this would probably be supporting something like the `Clonable` interface so you could do `query.clone()` to get a copy. I'll convert this issue into a feature request for this.

**Comment by preyesm** [comment]
Cool, thanks!
